### PR TITLE
Validation improvements

### PR DESCRIPTION
**Pull request by id2359**
- Exposed schema validation as a management command
- Corrected the json schema for Modjgo objects 
- Added test of command
- Prevented form model save with a missing Section object

### CODE REVIEW

File: rdrf/rdrf/management/commands/check_structure.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import yaml
+import jsonschema
+import errno
+import os
+
+explanation = "This command checks for schema validation errors"
+
+SCHEMA_LOCATIONS = ["/app/rdrf/rdrf/schemas/modjgo.yaml",
+                    "/env/src/django-rdrf/rdrf/rdrf/schemas/modjgo.yaml"]
+
```
Comment (sztamas) [inline rdrf/rdrf/management/commands/check_structure.py]: I don't like these docker container specific hardcoded paths here.
Can we do something about them?
For example get path based on a module's `__file__` or similar.

File: rdrf/rdrf/management/commands/check_structure.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    def handle(self, *args, **options):
+        self.test_mode = options.get("test_mode", False)
+        problem_count = 0
+        self.schema = self._load_schema()
+        registry_code = options.get("registry_code",None)
+        if registry_code is None:
+            self._print("Error: registry code required")
+            if not self.test_mode:
+                sys.exit(1)
```
Comment (sztamas) [inline rdrf/rdrf/management/commands/check_structure.py]: Not sure about this one, but it seems to me that you would want `sys.exit()` even if `test_mode is False`. Or does it make sense to continue at all if there is no registry code?

File: rdrf/rdrf/management/commands/check_structure.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        try:
+            registry_model = Registry.objects.get(code=registry_code)
+        except Registry.DoesNotExist:
+            self._print("Error: registry does not exist")
+            if not self.test_mode:
+                sys.exit(1)
```
Comment (sztamas) [inline rdrf/rdrf/management/commands/check_structure.py]: Same as comment at line 50.

File: rdrf/rdrf/management/commands/check_structure.py (hunk 1 above)
Comment (id2359) [inline rdrf/rdrf/management/commands/check_structure.py]: Yeah I was wondering about it too ok

File: rdrf/rdrf/management/commands/check_structure.py (hunk 2 above)
Comment (id2359) [inline rdrf/rdrf/management/commands/check_structure.py]: ok true

File: rdrf/rdrf/management/commands/check_structure.py (hunk 3 above)
Comment (id2359) [inline rdrf/rdrf/management/commands/check_structure.py]: Actually I did this on purpose - I'll have to remember why now   ,initially I tested for /expected sys exit in the tests, i'll have to re-try it  to prod my  memory

**Comment by id2359** [comment]
Ok I made the script locate the schema file without hardcoding paths

and also I've removed Test_mode and just tested for sys exit codes

**Pull request merged**